src/main/java/nextstep/subway/section/domain/SectionGroup.java

```diff
@@ -84,11 +101,89 @@ private void addTargetSection(int sourceSectionIndex, Section targetSection) {
 			sections.add(FIRST_INDEX, targetSection);
 			return;
 		}
-		sections.add(sourceSectionIndex, targetSection);
+	}
+
+	public void removeSectionByStation(Station removeTargetStation) {
+		validateRemoveSection();
+		removeTargetSectionWhenEdgeStation(removeTargetStation);
+		removeTargetSectionsWhenInnerStation(removeTargetStation);
+	}
+
+	private void removeTargetSectionsWhenInnerStation(Station targetStation) {
+		if (targetStation.isInnerStation(this.stations())) {
+			modifiedSection(targetStation);
+			removeSectionsRelatedWithTargetStation(targetStation);
+		}
+	}
```

Comment:
아래처럼 삭제할 구간만 제거해도 될 것 같은데요.
구간 삭제에 대한 단위 테스트도 작성하여 확인해보시면 좋을것 같아요 :)

```suggestion
	private void removeTargetSectionsWhenInnerStation(Station targetStation) {
		if (targetStation.isInnerStation(this.stations())) {
			modifiedSection(targetStation);
			sections.remove(targetStation);
		}
	}
```

Reply:
말씀하신 내용을 이해했습니다! 💡 
각 메서드는 종속적인 관계를 지니지 않고, 독립적인 수행을 해야하며,
`targetStation`과 연관된 `구간`을 모두 삭제 하지 않도록 해야된다는 말씀으로 이해했습니다. 🤔 

종전에 작성한 `modifiedSection()`, `removeSectionsRelatedWithTargetStation()` 메서드가 독립적으로 수행하는 기능이 아닌, 종속적인 관계를 가진 메서드로서 수행하고 있었습니다!
따라서 
1. `mofiedSection` -> `modifySections`로 메서드명을 고치고
2. 기존에 존재하던 하나의 `Section`을 수정, 다른 `Section`을 제거하는 내용을 `modifySections`에 포함하도록 하겠습니다.

메서드가 종속적인 관계를 끊는 것이 바람직할 것으로 생각합니다!
좋은 코멘트 감사합니다!! 👍 🙇